### ISSUE TITLE
Mismatch PSNR with papers on NeRF_Synthetic

### ISSUE THREAD

**Issue by renhaofan** [hbb1/2d-gaussian-splatting#158]
I train with all the default parameters with command:

`python train.py -s ./dataset/nerf_synthetic/$scene -m $output_dir/${scene} --eval`

The PSNR results are as follows:

| chair | drums  | ficus  | hotdog  | lego  | materials | mic  | ship  | mean  |
|------|------|------|------|------|------|------|------|------|
| 34.89 | 26.02 | 35.79| 36.84 | 33.80 | 30.15 | 33.99 | 31.10 | 32.83 |

To avoid the fluctuations problem mentioned in [#85](https://github.com/hbb1/2d-gaussian-splatting/issues/85), I conducted three experiments.  The PSNRs for the three experiments are 32.83, 32.79, and 32.83, respectively.

I noticed the same background error as mentioned in [#85](https://github.com/hbb1/2d-gaussian-splatting/issues/85).  Besides, I want to know how to reproduce the same results with paper.

![Screenshot_select-area_20240925091210](https://github.com/user-attachments/assets/b550514c-2862-40ef-b9fd-ccb79655fd8e)

**Comment by hbb1** [comment]
Our result reported in the paper is without using regularization. You default configuration uses regularization that decreases PSNR.

**Comment by hbb1** [comment]
Additionally, even using the regularization, the lego result should be  34.38

```python
python train.py -s ./data/nerf_synthetic/lego -m output/lego --eval 
python render.py -s data/nerf_synthetic/lego -m output/lego --eval  
python metrics.py -m output/lego/     
```

<img width="947" alt="image" src="https://github.com/user-attachments/assets/9f2fea56-139e-4be3-95a4-6b70cb86c0db">

How did you precisely reproduce the results?

**Comment by hbb1** [comment]
I have uploaded the nerf-evaluation script 7cb72dff1fa727633e3d078f503adfbf583fc39f, you can use the following commands to reproduce the results:

```bash
python scripts/nerf_eval.py    
python scripts/summary.py -m output/exp_nerf_synthetic
```
It will output the results as follow
```bash
Experiment      PSNR     SSIM    LPIPS
     chair 35.241447 0.985017 0.014486
     drums 26.090942 0.953859 0.039582
     ficus 35.387054 0.988067 0.011697
    hotdog 37.333900 0.984003 0.022910
      lego 35.139194 0.980221 0.019536
 materials 29.682457 0.957775 0.038627
       mic 34.848488 0.990276 0.007318
      ship 30.632690 0.904379 0.109990
Average PSNR: 33.04452180862427
Average SSIM: 0.9679495021700859
Average LPIPS: 0.03301838465267792
```

**Comment by renhaofan** [comment]
> Additionally, even using the regularization, the lego result should be 34.38
> 
> ```python
> python train.py -s ./data/nerf_synthetic/lego -m output/lego --eval 
> python render.py -s data/nerf_synthetic/lego -m output/lego --eval  
> python metrics.py -m output/lego/     
> ```
> 
> <img alt="image" width="947" src="https://private-user-images.githubusercontent.com/57170935/370539752-9f2fea56-139e-4be3-95a4-6b70cb86c0db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyNDMxNjAsIm5iZiI6MTcyNzI0Mjg2MCwicGF0aCI6Ii81NzE3MDkzNS8zNzA1Mzk3NTItOWYyZmVhNTYtMTM5ZS00YmUzLTk1YTQtNmI3MGNiODZjMGRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI1VDA1NDEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NWUyYjI5OWJmYjY2ZWU0OTQ1YzAwNDI3MmJjODU4N2JmNDc0NGY4ZjM2OWExMzVhMzc3N2M5OTdmYTYxZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XgBdCC6CeT_xOXpqJqH3oPDe0Smug-NBTr9D4s4OJtg">
> How did you precisely reproduce the results?

I use the totally same command with yours.
```
python train.py -s ./dataset/nerf_synthetic/$scene -m $output_dir/${scene} --eval
python render.py -s ./dataset/nerf_synthetic/$scene -m $output_dir/${scene} --eval
python metrics.py -m $output_dir/${scene}
```
I tried again: 

![Screenshot_select-area_20240925135828](https://github.com/user-attachments/assets/64cab06b-d1d3-43b7-bf82-dafa5efe29c5)

I'm confused about the result with the hyperparameter being the same.   I will use the following function to fix the random seed and try again.

```
def set_seed(seed=2024):
    import random
    import numpy as np
    if seed is not None:
        random.seed(seed)
        np.random.seed(seed)
        torch.manual_seed(seed)
        torch.cuda.manual_seed(seed)
        torch.cuda.manual_seed_all(seed)
        torch.backends.cudnn.benchmark = False
        torch.backends.cudnn.deterministic = True
    else: 
        print('No seed is specified, therefore seed is collected from system clock.')
```

**Comment by renhaofan** [comment]
> I have uploaded the nerf-evaluation script [7cb72df](https://github.com/hbb1/2d-gaussian-splatting/commit/7cb72dff1fa727633e3d078f503adfbf583fc39f), you can use the following commands to reproduce the results:
> 
> ```shell
> python scripts/nerf_eval.py    
> python scripts/summary.py -m output/exp_nerf_synthetic
> ```
> 
> It will output the results as follow
> 
> ```shell
> Experiment      PSNR     SSIM    LPIPS
>      chair 35.241447 0.985017 0.014486
>      drums 26.090942 0.953859 0.039582
>      ficus 35.387054 0.988067 0.011697
>     hotdog 37.333900 0.984003 0.022910
>       lego 35.139194 0.980221 0.019536
>  materials 29.682457 0.957775 0.038627
>        mic 34.848488 0.990276 0.007318
>       ship 30.632690 0.904379 0.109990
> Average PSNR: 33.04452180862427
> Average SSIM: 0.9679495021700859
> Average LPIPS: 0.03301838465267792
> ```

Now, I can reproduce the results. Thanks for your detailed reply.
```
Experiment      PSNR     SSIM    LPIPS
     chair 35.218410 0.984813 0.014820
     drums 26.097662 0.953851 0.039744
     ficus 35.376278 0.988051 0.011743
    hotdog 37.362965 0.984107 0.022657
      lego 35.047096 0.979762 0.020189
 materials 29.674755 0.957836 0.038601
       mic 34.732452 0.990147 0.007357
      ship 30.607735 0.904091 0.110266
Average PSNR: 33.01466917991638
Average SSIM: 0.9678320735692978
Average LPIPS: 0.033172034076415

```
